components/General/WrapWithFaram.js

```diff
@@ -0,0 +1,96 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import hoistNonReactStatics from 'hoist-non-react-statics';
+
+import Faram, { detachedFaram } from '#rscg/Faram';
+
+const propTypes = {
+    faramClassName: PropTypes.string,
+    schema: PropTypes.object.isRequired, // eslint-disable-line react/forbid-prop-types
+};
+
+const defaultProps = {
+    faramClassName: '',
+};
+
+
+export default ({ submitAction, submitCallback }) => (WrappedComponent) => {
+    class FaramWrappedComponent extends React.PureComponent {
+        static propTypes = propTypes;
+        static defaultProps = defaultProps;
+
+        constructor(props) {
+            super(props);
+
+            this.state = {
+                faramValues: {},
+                faramErrors: {},
+            };
+        }
+
+        handleFaramSuccess = (values) => {
+            const { [submitCallback]: onSubmit } = this.props;
+            if (onSubmit) {
+                onSubmit(values);
+            }
+        }
+
+        handleFaramFailure = (faramErrors) => {
+            this.setState({ faramErrors });
+        }
+
+        handleFaramChange = (faramValues, faramErrors) => {
+            this.setState({
+                faramValues,
+                faramErrors,
+            });
+        }
+
+        handleSubmit = () => {
+            detachedFaram({
+                value: this.state.faramValues,
+                schema: this.props.schema,
+                onValidationFailure: this.handleFaramFailure,
+                onValidationSuccess: this.handleFaramSuccess,
+            });
+        }
+
+        render() {
+            const {
+                schema,
+                [submitCallback]: onSubmit, // eslint-disable-line no-unused-vars
+                faramClassName,
+                ...otherProps
+            } = this.props;
+
+            const {
+                faramValues,
+                faramErrors,
+            } = this.state;
+
+            const newProps = { ...otherProps };
+            if (submitAction) {
```

Comment:
is submitAction a string identifier?
do we really need to inject this.handleSubmit?

Reply:
Yes

Reply:
Consider that we are wrapping a Button with this HOC. Then `submitAction` would be `onClick`.